features/storage/kvstore/filesystemstore/FileSystemStore.h

```diff
@@ -0,0 +1,229 @@
+     */
+    virtual int remove(const char *key);
+
+    /**
+     * @brief Start an incremental FileSystemStore set sequence.
+     *
+     * @param[out] handle               Returned incremental set handle.
+     * @param[in]  key                  Key.
+     * @param[in]  final_data_size      Final value data size.
+     * @param[in]  create_flags         Flag mask.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_start(set_handle_t *handle, const char *key, size_t final_data_size, uint32_t create_flags);
+
+    /**
+     * @brief Add data to incremental FileSystemStore set sequence.
+     *
+     * @param[in]  handle               Incremental set handle.
+     * @param[in]  value_data           value data to add.
+     * @param[in]  data_size            value data size.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_add_data(set_handle_t handle, const void *value_data, size_t data_size);
+
+    /**
+     * @brief Finalize an incremental FileSystemStore set sequence.
+     *
+     * @param[in]  handle               Incremental set handle.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int set_finalize(set_handle_t handle);
+
+    /**
+     * @brief Start an iteration over FileSystemStore keys.
+     *
+     * @param[out] it                   Returned iterator handle.
+     * @param[in]  prefix               Key prefix (null for all keys).
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_open(iterator_t *it, const char *prefix = NULL);
+
+    /**
+     * @brief Get next key in iteration.
+     *
+     * @param[in]  it                   Iterator handle.
+     * @param[in]  key                  Buffer for returned key.
+     * @param[in]  key_size             Key buffer size.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_next(iterator_t it, char *key, size_t key_size);
+
+    /**
+     * @brief Close iteration.
+     *
+     * @param[in]  it                   Iterator handle.
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    virtual int iterator_close(iterator_t it);
+
+private:
+
+    /**
+     * Build Full name of Key as a combination of FSST folder and key name
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    int _build_full_path_key(const char *key_src);
+
+    /**
+     * Verify Key file validity
+     *
+     * @returns 0 on success or a negative error code on failure
+     */
+    int _verify_key_file(const char *key, key_metadata_t *key_metadata, File *kv_file);
+
+private:
+    FileSystem *_fs;
+    PlatformMutex _mutex;
+
+    bool _is_initialized;
+    char _cfg_fs_path[FSST_PATH_NAME_SIZE + 1]; /* FileSystemStore path on FileSystem */
```

Comment:
It is probably better to declare these as pointers, allocate them at init and deallocate them at deinit. This way you can dynamically calculate path size (instead of limiting it to 16), so FSST_PATH_NAME_SIZE isn't required (instead you'll probably have a private path size member).

Reply:
You wrote "these" .. I can only do it with '_cfg_fs_path' that is fixed for the entire instance life - Do we still want to limit folder path size? (256 Bytes?)   
Allocating '_full_path_key' dynamically will be required at every set/get/remove.. operation for each different key file which is less preferable.

Reply:
You are right - "these" only referred to `_cfg_fs_path`.
You don't have to allocate it at every operation - only at init. As for the size, you just need to allocate space for <configured path size> + '/' + max name size (128) + '\0'.

Reply:
understood